### PR TITLE
Fix ESLint errors/warnings on linting-issues-repro branch

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
## What kind of change does this PR introduce?
Fix ESLint errors/warnings on `linting-issues-repro` branch (#2143)

## What is the current behavior?
ESLint errors/warnings
```sh
✖ 119 problems (98 errors, 21 warnings)
```

## What is the new behavior?
Less ESLint errors/warnings
```sh
✖ 11 problems (0 errors, 11 warnings)
```

## What steps did you take to test this?
Run `yarn lint`

## Checklist
- [ ] Documentation N/A
- [x] Testing 
- [x] Ready to be merged 
- [ ] Added myself to contributors table  N/A

### CODE REVIEW

**Comment by lbogdan** [comment]
@MichaelDeBoey Can we limit to **only** eslint errors / warnings, please?

**Comment by MichaelDeBoey** [comment]
@lbogdan Reverted the find/replace stuff

**Comment by MichaelDeBoey** [comment]
@lbogdan I still have a lot of these errors, but  can't figure out how to fix them

```sh
codesandbox-client/packages/app/src/app/components/Preview/DevTools/Problems/index.tsx
  5:8  error  Unable to resolve path to module 'codesandbox-api/dist/types/actions/correction'  import/no-unresolved
  5:8  error  Missing file extension for "codesandbox-api/dist/types/actions/correction"        import/extensions
  9:8  error  Unable to resolve path to module 'codesandbox-api/dist/types/actions/error'       import/no-unresolved
  9:8  error  Missing file extension for "codesandbox-api/dist/types/actions/error"             import/extensions
```

File: packages/app/src/app/components/Preview/DevTools/Tabs/index.tsx [hunk 1]
```diff
@@ -69,7 +69,7 @@ const DevToolTabs = ({
               devToolIndex={devToolIndex}
               moveTab={moveTab}
               index={i}
-              key={i}
+              key={pane.id}
```
Comment (CompuIves) [inline packages/app/src/app/components/Preview/DevTools/Tabs/index.tsx]: This should be `i` (with an eslint ignore), since `pane.id` is not unique. Leads to small bugs.

File: packages/app/src/app/components/Preview/DevTools/Tests/TestElement/elements.ts [hunk 2]
```diff
@@ -22,8 +22,7 @@ export const Actions = styled.div`
 `;
 
 export const TestName = styled.div`
-  padding: 0.25rem;
-  padding-left: 20px;
+  padding: 0.25rem 0.25rem 0.25rem 20px;
```
Comment (CompuIves) [inline packages/app/src/app/components/Preview/DevTools/Tests/TestElement/elements.ts]: In general I'd like the explicit separate rule for `padding-left`. I don't mind this change, but I don't think future usages of `padding-left` etc need to be refactored.

File: packages/app/src/app/components/Preview/DevTools/Tests/TestElement/elements.ts [hunk 3]
```diff
@@ -87,88 +86,84 @@ export const FileData = styled.div`
...
-    background-color: ${props =>
-      !props.theme.light ? 'rgba(0, 0, 0, 0.2)' : 'rgba(255, 255, 255, 0.2)'};
-    color: ${props =>
-      props.theme.light ? 'rgba(0, 0, 0, 1)' : 'rgba(255, 255, 255, 1)'};
-    border-left-color: ${props => props.theme.secondary.clearer(0.5)};
-
-    ${Actions} {
-      opacity: 1;
-    }
+  ${({ selected, theme }) => css`
```
Comment (CompuIves) [inline packages/app/src/app/components/Preview/DevTools/Tests/TestElement/elements.ts]: Are these just style changes? I'm not sure whether I'm comfortable merging this since it can fundamentally change style behaviour and we're not working on this part of the codebase right now. I'd prefer to revert the style changes here.

File: packages/app/src/app/components/Preview/DevTools/Tests/index.tsx [hunk 4]
```diff
@@ -25,24 +23,26 @@ export type Status = 'idle' | 'running' | 'pass' | 'fail';
...
+        }[]
+      | null;
+  }[];
 };
 
 export type Test = {
-  testName: Array<string>;
-  duration: number | undefined;
+  testName: string[];
+  duration?: number;
```
Comment (CompuIves) [inline packages/app/src/app/components/Preview/DevTools/Tests/index.tsx]: Let's keep this `duration: number | undefined`. Like I said before, it changes whether the key is set on the object or not (in this case it's always set, but can be `undefined`);

```js
const a = { test: undefined }; // `('test' in a) === true
const a = { }; // `('test' in a) === false
```

File: packages/app/src/app/components/Preview/DevTools/Tests/TestElement/elements.ts (hunk 3 above)
Comment (MichaelDeBoey) [reply packages/app/src/app/components/Preview/DevTools/Tests/TestElement/elements.ts]: These are fixes for having a double use of the `props` name as a variable.

File: packages/app/src/app/components/Preview/DevTools/Tests/index.tsx [hunk 5]
```diff
@@ -25,24 +23,26 @@ export type Status = 'idle' | 'running' | 'pass' | 'fail';
...
     _originalFunctionName: string;
     _originalColumnNumber: number;
     _originalLineNumber: number;
-    _originalScriptCode: Array<{
-      lineNumber: number;
-      content: string;
-      highlight: boolean;
-    }> | null;
-  }>;
+    _originalScriptCode:
```
Comment (CompuIves) [inline packages/app/src/app/components/Preview/DevTools/Tests/index.tsx]: In this case I prefer using `Array`, so `Array` when using object literals, `[]` for types/interfaces.

File: packages/app/src/app/vscode/extensionHostWorker/services/searchService.ts [hunk 6]
```diff
@@ -11,15 +11,6 @@ initializeGlobals();
 
 async function initialize() {
   await initializeBrowserFS({ syncSandbox: true, syncTypes: true });
-
-  self.addEventListener('message', e => {
```
Comment (CompuIves) [inline packages/app/src/app/vscode/extensionHostWorker/services/searchService.ts]: Why is this removed?

Comment (MichaelDeBoey) [reply packages/app/src/app/vscode/extensionHostWorker/services/searchService.ts]: The listener wasn't doing anything

```ts
self.addEventListener('message', e => {	
    if (e.data.$type === 'input-write') {	
      const { $type, $data } = e.data.$data;	
      if ($type === 'file-search') {	
        const { query } = $data;	
      }	
    }	
  });
```

File: packages/app/src/app/vscode/index.ts [hunk 7]
```diff
@@ -74,9 +75,9 @@ class VSCodeManager {
         this.controller.getSignal('editor.toggleEditorPreviewLayout')();
       },
       keybindings: {
-        primary: KeyMod.Shift | KeyMod.Alt | KeyCode.KEY_9,
+        primary: KeyMod.Shift || KeyMod.Alt || KeyCode.KEY_9,
```
Comment (CompuIves) [inline packages/app/src/app/vscode/index.ts]: This has to be a `|`, keyboard shortcuts use binary shifting to determine the shortcut. If we use `||` it will break the shortcut. We should change our eslint rules for this.

File: packages/app/src/app/vscode/initializers.ts [hunk 8]
```diff
@@ -106,7 +106,7 @@ export function setVimExtensionEnabled(vimEnabled: boolean) {
   } else {
     // Auto disable vim extension
     if (
-      localStorage.getItem('vs-global://extensionsIdentifiers/disabled') ==
+      localStorage.getItem('vs-global://extensionsIdentifiers/disabled') ===
```
Comment (CompuIves) [inline packages/app/src/app/vscode/initializers.ts]: This has to be `==`, to also account for `null` values.

File: packages/node-services/src/child_process.ts [hunk 9]
```diff
@@ -34,15 +34,15 @@ class Stream extends EventEmitter {
     super();
   }
 
-  setEncoding(encoding: string) {}
+  setEncoding() {}
 
-  write(message: string, encoding: string) {
+  write(message: string) {
```
Comment (CompuIves) [inline packages/node-services/src/child_process.ts]: Are we required to remove these parameters? I think they have a documentation purpose. If linting requires it it's okay to remove them.

File: packages/node-services/src/module.ts [hunk 10]
```diff
@@ -21,12 +21,12 @@ function updateChildren(
 export default class Module {
   id: string;
   exports: any;
-  parent: Module | undefined;
+  parent?: Module;
```
Comment (CompuIves) [inline packages/node-services/src/module.ts]:
```suggestion
  parent: Module | undefined;
```

File: packages/node-services/src/module.ts [hunk 11]
```diff
@@ -84,7 +84,7 @@ export default class Module {
   }
 
   _compile(content: string, filename: string) {
-    var _self = this;
+    const _self = this;
```
Comment (CompuIves) [inline packages/node-services/src/module.ts]:
```suggestion
    var _self = this;
```

This is pretty important. This should not be a block variable, this is copied from the node code.

File: packages/app/src/app/store/providers/FSSync.ts [hunk 12]
```diff
@@ -90,9 +90,9 @@ async function syncDependencyTypings(
           sendTypes();
         } catch (e) {
           if (process.env.NODE_ENV === 'development') {
+            // eslint-disable-next-line no-console
             console.warn('Trouble fetching types for ' + depName);
           }
-          return {};
```
Comment (CompuIves) [inline packages/app/src/app/store/providers/FSSync.ts]: We should return an empty object here.

File: packages/node-services/src/child_process.ts (hunk 9 above)
Comment (MichaelDeBoey) [reply packages/node-services/src/child_process.ts]: They trigger the `no-undefined` rule

File: packages/app/src/app/vscode/initializers.ts (hunk 8 above)
Comment (MichaelDeBoey) [reply packages/app/src/app/vscode/initializers.ts]: Will make it an
```ts
[null, undefined].includes( /* ... */)
```
instead so we don't have to disable the linting

Comment (CompuIves) [reply packages/app/src/app/vscode/initializers.ts]: I think the rule has an exception for `== null`, I'm not 100% sure though!

File: packages/app/src/app/vscode/extensionHostWorker/services/searchService.ts (hunk 6 above)
Comment (CompuIves) [reply packages/app/src/app/vscode/extensionHostWorker/services/searchService.ts]: That's true, I'd prefer to keep it there since I was planning on using this for the search implementation.

Comment (MichaelDeBoey) [reply packages/app/src/app/vscode/extensionHostWorker/services/searchService.ts]: OK I'll revert it and just disable the following line then 👍
```ts
const { query } = $data;
```

File: packages/app/src/app/components/Preview/DevTools/Tests/index.tsx (hunk 4 above)
Comment (MichaelDeBoey) [reply packages/app/src/app/components/Preview/DevTools/Tests/index.tsx]: I understand the difference when you test those 2, but if I put  `duration?: number`, WebStorm gives me `duration:  number | undefined`, so these 2 are the same I think.

File: packages/app/src/app/vscode/initializers.ts (hunk 8 above)
Comment (MichaelDeBoey) [reply packages/app/src/app/vscode/initializers.ts]: But you're checking on `undefined` here so the rule errors out.

Doing both a check to `null` and `undefined` is the same as doing the
```ts
[null, undefined].includes(/* ... */)
```
Which is a bit cleaner than the double check I think, no? 🤔

**Comment by MichaelDeBoey** [comment]
@CompuIves I resolved all discussions 🙂

**Comment by MichaelDeBoey** [comment]
Always a pleasure 🙂
